File: app/src/main/java/edu/luc/etl/cs313/misc/boundedcounter/cli/BoundedCounter.java
package edu.luc.etl.cs313.misc.boundedcounter.cli;

/** A bounded counter abstraction. */
// begin-interface-BoundedCounter
public interface BoundedCounter {
    void increment();
    void decrement();
    int get();
    boolean isFull();
    boolean isEmpty();
}
// end-interface-BoundedCounter

File: app/src/main/java/edu/luc/etl/cs313/misc/boundedcounter/cli/BoundedCounter.kt
package edu.luc.etl.cs313.misc.boundedcounter.cli

/** A bounded counter abstraction.  */
// begin-interface-BoundedCounter
interface BoundedCounter {
    fun increment()
    fun decrement()
    fun get(): Int
    val isFull: Boolean
    val isEmpty: Boolean
}
// end-interface-BoundedCounter
